Improve "Foreman on AWS" guide (#3485)
theforeman/foreman-documentation: guides/common/modules/con_prerequisites-for-foreman-on-aws.adoc

[id="prerequisites-for-{project-context}-on-aws"]
= Prerequisites for {Project} on AWS

> Installing and running {ProjectServer} and {SmartProxyServers} on Amazon Web Services (AWS) has additional requirements to your environment.

Reviewer: You might not even need this sentence.

Reviewer: Or perhaps be more specific instead:
```suggestion
In additional to the standard prerequisites for installing and running {ProjectServer} and {SmartProxyServers}listed in <links to prereqs>, ensure to also meet the following prerequisites specific to deploying on Amazon Web Services (AWS):
```

This is way longer but also much more specific than the rather vague wording about "additional requirements".

* Ensure that the Amazon EC2 instance type meets or exceeds the {InstallingServerDocURL}system-requirements_{project-context}[System requirements] in _{InstallingServerDocTitle}_.

> For the best performance, use an https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/storage-optimized-instances.html[AWS storage optimized instance].

Reviewer: ```suggestion
For the best performance, use an link:https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/storage-optimized-instances.html[AWS storage optimized instance].
```
Or maybe even (nitpick trigger warning):
```suggestion
See also an link:https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/storage-optimized-instances.html[AWS storage optimized instance] for further guidance.
```
Can we really promise that the guidelines will ensure "best" performance?

ifndef::foreman-deb[]

> * Use {InstallingServerDocURL}satellite-storage-requirements_{project-context}[Storage requirements] in _{InstallingServerDocTitle}_ to understand and assign the correct storage to your AWS EBS volumes.

Reviewer: Link doesn't lead to the Storage requirements section.

Reviewer: Is there a way to link to the Install prerequisites only once, instead of twice (System reqs and Storage reqs)? It would be easier to maintain one high-level link than two deep-level links.

* Store content on an EBS volume that is separate to the boot volume.
* Mount your content EBS volume separately in the operating system.
* Optional: Store other data on a separate EBS volume.
endif::[]
* If you want {ProjectServer} and {SmartProxyServer} to communicate using external DNS hostnames, open the required ports for communication in the AWS Security Group that is associated with the instance.

> .Amazon Web Service requirements

Reviewer: It's weird to see this one list of requirements in gerund. Does it mean that I need to do these actions ("Create and access[...]"), that Foreman must be able to do them ("Foreman must be able to create and access[...]"), or that AWS itself needs to be able to do them ("AWS must be able to create[...]")? I know that some of these don't even make sense, but it would be best to clarify by using direct, imperative language that makes it clear who is supposed to do what.

ifdef::foreman-deb[]
* Creating and accessing {DL} images in AWS
endif::[]
ifndef::foreman-deb[]
* Creating and accessing {EL} images in AWS
endif::[]
* Editing network access in AWS Security
* Creating EC2 instances and EBS volumes
* Launching EC2 instances
* Importing and exporting virtual machines in AWS
* Using AWS Direct Connect

.System requirements
* To install {ProjectServer} in your AWS environment, ensure that it meets {ProjectServer} system requirements.
For more information, see {InstallingServerDocURL}system-requirements_{project-context}[System requirements] in _{InstallingServerDocTitle}_.
* To install {SmartProxyServer} in your AWS environment, ensure that it meets {SmartProxyServer} system requirements.

> For more information, see {InstallingSmartProxyDocURL}system-requirements_{smart-proxy-context}[System requirements] in _{InstallingSmartProxyDocTitle}_.

Reviewer: This links to the Storage and System prereqs again. See my point above about perhaps finding a way to link to them only once.

* For more information about Amazon Web Services and terminology, see https://docs.aws.amazon.com/ec2/[Amazon Elastic Compute Cloud Documentation].

> * For more information about Amazon Web Services Direct Connect, see https://docs.aws.amazon.com/directconnect/latest/UserGuide/Welcome.html[What is AWS Direct Connect?].

Reviewer: These are not prerequisites. Can you move this somewhere to Additional resources?

[id="prerequisites-for-{project-context}-on-aws"]
= Prerequisites for {Project} on AWS

Installing and running {ProjectServer} and {SmartProxyServers} on Amazon Web Services (AWS) has additional requirements to your environment.

.Amazon Web Service requirements
ifndef::foreman-deb[]
* Use {InstallingServerDocURL}storage-requirements_{project-context}[Storage requirements] in _{InstallingServerDocTitle}_ to understand and assign the correct storage to your AWS EBS volumes.
See also an link:https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/storage-optimized-instances.html[AWS storage optimized instance] for further guidance.

> * Store content on an EBS volume that is separate to the boot volume.
> * Mount your content EBS volume separately in the operating system.

Reviewer 2: I'm struggling a bit with the formulation here. Although I understand what these bullet points are saying, AWS typically refers to the root volume

>  The root volume contains the image used to boot the instance. Each instance has a single root volume. You can add storage volumes to your instances during or after launch.

The point about mounting reads a bit like "Don't mount an EBS volume to /". Assuming you're refering to pulp content we could state the target directory or even become more abstract:

```suggestion
* Create EBS volumes for directories expected to contain larger amounts of data like `/var/lib/pulp` and ensure they are correctly mounted on start-up and before continuing the installation.
```

Also, this is no hard requirement although it's listed in the requirements block. You could simply grow the root volume which I would, however, not recommend.

Author: Applied your suggestion.
